### DISCUSSION TITLE
Look at template Haskell again?

### DISCUSSION THREAD

**Discussion by yaxu** [tidalcycles/Tidal#1122] [General]
I don't fully understand TH, but a couple of potential uses for it:

Parsing the mininotation at compile time, to reduce/remove runtime errors. I've had a couple of cases where errors have turned up late, and patterns have rolled back unexpectedly. We could do some pre-processing of the source to still support the mininotation-as-strings (feedforward already does something similar to track source code location of the mininotation for highlighting active events, in a hacky way). This would need to be optional so that minitidal isn't affected.

Removing the large amount of boilerplate from `Sound.Tidal.Parse`, especially as I want to add a variant of each control function that accepts a bus number. Would be nice although if minitidal can't work with template haskell (is that the case?) then that would be a point against.

**Comment by sss-create** [comment]
I have no experience with TH and therefore would be in favor of not using it.
